includes/http/WP_AI_Client_Discovery_Strategy.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * WordPress AI Client Discovery Strategy
+ *
+ * @package WordPress\AI_Client
+ * @since n.e.x.t
+ */
+
+namespace WordPress\AI_Client\HTTP;
+
+use Http\Discovery\Psr18ClientDiscovery;
+use Http\Discovery\Strategy\DiscoveryStrategy;
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Psr\Http\Client\ClientInterface;
+
+/**
+ * Discovery strategy for WordPress HTTP client.
+ *
+ * @since n.e.x.t
+ */
+class WP_AI_Client_Discovery_Strategy implements DiscoveryStrategy {
+
+	/**
+	 * Initialize and register the discovery strategy.
+	 *
+	 * @since n.e.x.t
+	 * @return void
+	 */
+	public static function init() {
+		// Check if discovery is available.
+		if ( ! class_exists( '\Http\Discovery\Psr18ClientDiscovery' ) ) {
+			return;
+		}
+
+		// Register our discovery strategy.
+		Psr18ClientDiscovery::prependStrategy( self::class );
+	}
+
+	/**
+	 * Get candidates for discovery.
+	 *
+	 * @param string $type The type of discovery.
+	 *
+	 * @return array<array<string, mixed>>
+	 */
+	public static function getCandidates( $type ) {
+		// PSR-18 HTTP Client.
+		if ( ClientInterface::class === $type ) {
+			return array(
+				array(
+					'class'     => array( __CLASS__, 'createWordPressClient' ),
+					'condition' => array(
+						WordPress_HTTP_Client::class,
+						Psr17Factory::class,
+					),
+				),
+			);
+		}
+
+		// PSR-17 factories - Nyholm's Psr17Factory implements all of them.
+		$psr17_factories = array(
```

Comment:
This is mostly a question for my own understanding: Why do we need this? Why wouldn't we be able to _only_ inject our WordPress HTTP Client and leave the rest?

Reply:
So our Client is returned above when the PSR-18 ClientInterface is being asked for. When a PSR-17 factory is asked for, we use the Nyholm one as we don't need to create a custom one for generating PSR Messages and such.